doc/NAT/SAI-NAT-API.md

```diff
@@ -0,0 +1,443 @@
+
+## 2.0 Configuring NAT zones
+NAT zones are configured for translation. Only when packet crosses across NAT zones then NAT translation is done. SONiC will configure interfaces as a member of the zones. Zone ID is passed in the API setting up NAT rules.
+
+A new uint8_t Zone ID attribute is added in sai_router_interface_attr_t.
+> “SAI_ROUTER_INTERFACE_NAT_ZONE_ID”
+> set_router_interface_attribute() API is used to set the NAT Zone ID.
+
+## 3.0 Enabling NAT in switch
+NAT feature is enabled at the switch level. SAI driver must handle this attribute and set corresponding HW register. Some of the settings can be
+-	Global config register
+-	Control settings for NAT miss packets for dynamic NAT
+
+A new boolean field is added to sai_switch_attr in saiswitch.h
+> “SAI_SWITCH_ATTR_NAT_ENABLE”
+> set_switch_attribute() API is used to enable/disable NAT feature.
+
+## 4.0 Enable Traps for SNAT and DNAT Miss Packets
+Following two traps are added to hostif. Hostif driver MUST enable these traps for SNAT and DNAT miss in hw for receiving the packets. 
+SNAT/DNAT/HARIPIN miss packets are received on a regular netdev channel to the application.
+
+- “SAI_HOSTIF_TRAP_TYPE_SNAT_MISS”
+- “SAI_HOSTIF_TRAP_TYPE_DNAT_MISS”
+- "SAI_HOSTIF_TRAP_TYPE_NAT_HAIRPIN"
+
+## 5.0 NAT Object Workflow
+ NAT or Network Address translation is used to avoid exposing the Internal or private IP addresses when the Host systems in an enterprise network or Datacenter communicate with external internet servers . The main advantage of using NAT is to reduce the usage of the Public IP addresses allocated for an enterprise or Data center.NAT feature allows the capability to change the Src IP address , Destination IP address , Src TCP/UDP port number or Destination TCP/UDP port number for an IP packet undergoing L3  routing in the switch.
+
+### 5.1 Symmetric SNAT and DNAT
+For directional symmetric flows there is always a corresponding DNAT entry for each SNAT entry. SAI APIs do not allow implicit installation of reverse direction NAT entry. SAI API MUST be invoked for both SNAT and DNAT entry to be installed.
+
+> From Zone ID: 100
+To Zone ID: 200
+VRF: None
+Packet Count: Enable
+Byte Count: Enable
+ExternalEndpoint: 65.55.42.1:1024 
+InternalEndpoint: 10.0.0.1:6000
+
+##### Step 1: Create a Source and Destination NAT Entry object:
+```sh
+sai_attribute_t nat_entry_attr[10];
+nat_entry_t snat_entry;
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_NAT_TYPE;
+nat_entry_attr[0].value = SAI_NAT_TYPE_SOURCE_NAT;
+
+nat_entry_attr[1].id = SAI_NAT_ENTRY_ATTR_SRC_IP;
+nat_entry_attr[1].value.u32 = 65.55.42.1; //example string
+
+nat_entry_attr[2].id = SAI_NAT_ENTRY_ATTR_L4_SRC_PORT;
+nat_entry_attr[2].value.u16 = 1024;
+
+nat_entry_attr[3].id = SAI_NAT_ENTRY_ATTR_TO_ZONE;
+nat_entry_attr[3].value.u32 = 200;

```

Comment:
This attribute no longer exists, should be deleted

Reply:
removed